catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -367,9 +371,67 @@ protected SolrQuery postAdapt(
 
     setSortProperty(request, query, filterDelegate);
 
+    filterAttributes(request, query);
+
     return query;
   }
 
+  private void filterAttributes(QueryRequest request, SolrQuery query) {
+    if (skipFilteredAttributes(request)) {
+      return;
+    }
+
+    Set<String> excludedAttributes = (Set<String>) request.getPropertyValue(EXCLUDE_ATTRIBUTES);
+
+    Set<String> excludedFields =
+        resolver
+            .fieldsCache
+            .stream()
+            .filter(Objects::nonNull)
+            .filter(field -> excludedAttributes.stream().anyMatch(field::startsWith))
+            .collect(Collectors.toSet());
+
+    Set<String> wildcardFields =
+        SchemaFields.FORMAT_TO_SUFFIX_MAP
+            .values()
+            .stream()
+            .filter(suffix -> excludedFields.stream().noneMatch(field -> field.endsWith(suffix)))
+            .map(suffix -> "*" + suffix)
+            .collect(Collectors.toSet());
+
+    Set<String> includedFields =
+        SchemaFields.FORMAT_TO_SUFFIX_MAP
+            .values()
+            .stream()
+            .filter(suffix -> excludedFields.stream().anyMatch(field -> field.endsWith(suffix)))
+            .flatMap(
+                suffix ->
+                    resolver
+                        .fieldsCache
+                        .stream()
+                        .filter(Objects::nonNull)
+                        .filter(field -> field.endsWith(suffix))
+                        .filter(field -> excludedAttributes.stream().noneMatch(field::startsWith)))
+            .collect(Collectors.toSet());
+
+    Set<String> fields = Sets.union(includedFields, wildcardFields);
+
+    if (query.getFields() != null && query.getFields().length() > 2) {
+      fields = Sets.union(fields, Sets.newHashSet(query.getFields().substring(2).split(",")));
+    }
+
+    if (!fields.isEmpty()) {
+      query.setFields(fields.toArray(new String[fields.size()]));
+    }
+  }
+
+  private boolean skipFilteredAttributes(QueryRequest request) {
+    return !request.hasProperties()
+        || !request.containsPropertyName(EXCLUDE_ATTRIBUTES)
+        || !(request.getPropertyValue(EXCLUDE_ATTRIBUTES) instanceof Set)
+        || "true".equals(System.getProperty("solr.client.filterAttributes.disable"));
```

Comment:
❓ Might we also add one additional check about the set and ensure that it is also not empty? This would ensure none of the `filterAttributes` logic runs unecessarily.